ci/mage/engine.go

```diff
@@ -163,7 +167,7 @@ func (t Engine) Dev(ctx context.Context) error {
 		"--privileged",
 	}...)
 
-	runArgs = append(runArgs, imageName, "--debug", "--debugaddr=0.0.0.0:6060")
+	runArgs = append(runArgs, imageName, "--extra-debug", "--debugaddr=0.0.0.0:6060")
```

Comment:
Yeah, I think there's a lot of upside to enabling this for `./hack/dev` and basically no downside, since these logs don't end up in CI. Without enabling this I'm basically helpless when debugging OTel issues.

Not sure what'll happen to this when we retire Mage, but I can burn that bridge when I get to it.